homeassistant/components/izone/discovery.py

```diff
@@ -19,11 +19,14 @@
 class DiscoveryService(pizone.Listener):
     """Discovery data and interfacing with pizone library."""
 
-    def __init__(self, hass):
+    def __init__(self, hass: HomeAssistant) -> None:
         """Initialise discovery service."""
         super().__init__()
         self.hass = hass
-        self.pi_disco = None
+        hass.data[DATA_DISCOVERY_SERVICE] = self
```

Comment:
I think the previous design was a bit cleaner.  Ideally we could avoid teaching this class about `hass`

Reply:
This class exists to dispatch things through the hass dispatcher, it's a bit hard for it not to know about hass